File: tasks/javascript/8kyu/SimpleMultiplication.js
// Simple multiplication

// This kata is about multiplying a given number by eight if it is an even number and by nine otherwise.

// Sample Tests:
// const chai = require("chai");
// const assert = chai.assert;
// chai.config.truncateThreshold=0;

// describe("Basic Tests", () => {
//   it("Testing for fixed tests", () => {
//     assert.strictEqual(simpleMultiplication(2),16,'Should return given argument times eight...')
//     assert.strictEqual(simpleMultiplication(1),9, 'Should return given argument times nine...')
//     assert.strictEqual(simpleMultiplication(8),64,'Should return given argument times eight...')
//     assert.strictEqual(simpleMultiplication(4),32,'Should return given argument times eight...')
//     assert.strictEqual(simpleMultiplication(5),45,'Should return given argument times nine...')
//   });
// });

function simpleMultiplication(number) {
  return number % 2 === 0 ? number * 8 : number * 9;
}

File: tasks/cpp/8kyu/Simple_multiplication.cpp
/*
This kata is about multiplying a given number by eight if it is an even number and by nine otherwise.
  Describe(simpleMultiplicationTest)
{
    It(multiply)
    {
        Assert::That(simpleMultiplication(2), Equals(16));
        Assert::That(simpleMultiplication(1), Equals(9));
        Assert::That(simpleMultiplication(8), Equals(64));
        Assert::That(simpleMultiplication(4), Equals(32));
        Assert::That(simpleMultiplication(5), Equals(45));
    }
};
*/
int simpleMultiplication(int a){
    return (a % 2 == 0) ? 8 * a : 9 * a; 
}
